sources/views/contact-form.js

```diff
@@ -0,0 +1,247 @@
+									name: "Company",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Website",
+									name: "Website",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Address",
+									name: "Address",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								}
+							]
+						},
+						{
+							margin: 20,
+							rows: [
+								{
+									view: "text",
+									label: "Email",
+									name: "Email",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Skype",
+									name: "Skype",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Phone",
+									name: "Phone",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "datepicker",
+									label: "Birthday",
+									name: "Birthday",
+									format: webix.i18n.longDateFormatStr,
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									cols: [
+										{
+											type: "clean",
+											localId: "contactPhoto",
+											name: "Photo",
+											height: 200,
+											gravity: 2,
+											template: contact => `<div class="user-info-photo">
+											<img src=${contact.Photo || noPhoto} alt="User photo" /></div>`
+										},
+										{
+											rows: [
+												{},
+												{
+													view: "uploader",
+													localId: "photoUploader",
+													label: "Change photo",
+													accept: "image/png, image/jpg, image/jpeg",
+													autosend: false,
+													css: "button--style"
+												},
+												{
+													view: "button",
+													label: "Delete photo",
+													css: "button--style",
+													click: () => this.deletePhoto()
+												}
+											]
+										}
+									]
+								}
+							]
+						}
+					]
+				},
+				{},
+				{
+					cols: [
+						{view: "button", label: "Cancel", css: "button--style", click: () => this.closeForm()},
+						{
+							view: "button",
+							label: "",
+							css: "button--style",
+							localId: "actionsButton",
+							click: () => this.addEditContact()
+						}
+					]
+				}
+			]
+		};
+		return form;
+	}
+
+	init() {
+		this.form = this.$$("contactForm");
+		this.contactPhoto = this.$$("contactPhoto");
+		const fileReader = new FileReader();
+
+		this.$$("photoUploader").attachEvent("onBeforeFileAdd", (data) => {
+			fileReader.readAsDataURL(data.file);
+			fileReader.onload = () => this.contactPhoto.setValues({Photo: fileReader.result});
```

Comment:
You can set this handler once (right after declaring `fileReader`)

Reply:
Fixed.